### PR TITLE
Replace vega-lib with taucharts

### LINKED ISSUE #563: Improve the Y axis labels for the tests in the history report
#### Problem description

> A small but relevant comment from me and Laura - could the Y axis labels for the tests in the report be a little more specific per test to avoid misinterpretation of the metric?
@JenBartell on #561

### PR DESCRIPTION
**Pull request by ChristianLieven**
* [x] fix #563, #627
* [x] add an entry to the [next release](../HISTORY.rst)

* Replace vega with taucharts in all reports.
* Fix plots to be large enough.
* Fix responsive behaviour. 
* Add clear axis labels and understandable titles.
* Add an option to save all plots to png, json and csv.
* Color code the category plots just like the results buttons 
  on the snapshot report.
* Fix the data that is displayed in the independent an specific sections 
  of the history report.
* Invert metrices and display as percentages.
* Add hover-over tooltips for all plots.
* Implement lazy loading for all the reports, meaning that the content of 
  an expansion panel in any of the reports is only rendered when it is open.

Screenshot 2019-04-04 at 16 34 08

Screenshot 2019-04-04 at 16 33 25

### CODE REVIEW

File: memote-report-app/src/app/tauchart-history/tauchart-history.component.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  template: '<div id="tau-history-{{css_tag}}" style="height: 50vh; width: 40vw"> </div>',
+  encapsulation: ViewEncapsulation.None
+})
+export class TauChartHistoryComponent implements OnInit, AfterViewInit {
+  @Input() testObject: any;
+  @Input() testId: string;
+  format_type: string;
+  chart: any;
+  historyData: Object[];
+  css_tag = 'total';
```
Comment (kvikshaug) [inline memote-report-app/src/app/tauchart-history/tauchart-history.component.ts]: I'd maybe call this a `unique_id` or similar instead - looks to me like it's not really used for css, just as a unique html id attribute.

File: memote-report-app/src/index.html [hunk 2]
```diff
@@ -10,6 +10,7 @@
   <script type="text/javascript" async
   src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML">
 </script>
+<link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/npm/taucharts@2/dist/taucharts.min.css">
```
Comment (kvikshaug) [inline memote-report-app/src/index.html]: I'm a bit skeptical to requiring external resources in the report, but I realize that wasn't introduced in this PR though.

File: memote-report-app/src/app/tauchart-history/tauchart-history.component.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+  constructor(private data: ReportDataService) {}
+
+  public initialize() {
+    this.chart.renderTo('#tau-history-' + this.css_tag);
+  }
+ 
+  public invertScoredData(history: Object[]) {
```
Comment (kvikshaug) [inline memote-report-app/src/app/tauchart-history/tauchart-history.component.ts]: Why doesn't the score come inverted in the results in the first place?

File: memote-report-app/src/app/tauchart-history/tauchart-history.component.ts (hunk 1 above)
Comment (ChristianLieven) [reply memote-report-app/src/app/tauchart-history/tauchart-history.component.ts]: Good call! It is indeed.

**Pull request merged**